service/history/workflowTaskHandlerCallbacks.go

```diff
@@ -548,6 +548,9 @@ func (handler *workflowTaskHandlerCallbacksImpl) handleWorkflowTaskCompleted(
 		newMutableState = nil
 
 		if wtFailedCause.workflowFailure != nil {
+			// Flush buffer event before failing the workflow
+			ms.FlushBufferedEvents()
+
 			attributes := &commandpb.FailWorkflowExecutionCommandAttributes{
 				Failure: wtFailedCause.workflowFailure,
 			}
```

Comment:
i believe there is a util which flush & fail the workflow, can we plz use that util?